src/store/modules/projects/index.js

```diff
@@ -186,6 +186,25 @@ const actions = {
     }
   },
 
+  /**
+   * Send request to generate new integration token for project
+   *
+   * @param {Function} commit - standard Vuex commit function
+   * @param {String} projectId - project id for generation new token
+   *
+   * @returns {Promise<void>}
+   */
+  async [GENERATE_NEW_INTEGRATION_TOKEN]({ commit }, { projectId }) {
+    const response = await projectsApi.generateNewIntegrationToken(projectId);
+
+    const updatedProject = response.data.generateNewIntegrationToken.record;
+
+    console.log(updatedProject);
+    if (updatedProject) {
+      commit(mutationTypes.UPDATE_PROJECT, updatedProject);
```

Comment:
this will override all the Project fields?

Reply:
No, only token

Reply:
(in this case)